### DISCUSSION TITLE
After Update from 2 to 3 - where are the secrets for devices

### DISCUSSION THREAD

**Discussion by greinick** [netbox-community/netbox#7362] [Q&A]
Hi,

I hope, I missed an easy step after updating from the most recent netbox 2 to version 3.0.3. Where are the secrets for devices hiding? :)

Fingers crossed they still exist.

Thanks for hints and help. Götz

**Comment by greinick** [comment]
oh .... from a release note I found:

_The secrets functionality present in prior releases of NetBox has been removed. The NetBox maintainers strongly recommend the adoption of Hashicorp Vault in place of this feature. Development of a NetBox plugin to replace the legacy secrets functionality is also underway._

Is there a chance for recovering the secrets without restoring the netbox from a backup?

**Reply by candlerb** [reply]
The tables containing the secrets are still there.  A future secrets plugin may use them.

**Reply by greinick** [reply]
oh thanks, sounds good. Any "workaround" or suggestion how I can access the secrets right now? In an decrypted way :).

**Reply by candlerb** [reply]
Sorry, not off the top of my head: I don't have a Netbox instance with secrets in it to see what format the data is in.  Maybe keeping an old instance of netbox2 around for this purpose is reasonable.

**Reply by DanSheps** [reply]
https://github.com/dansheps/netbox-secretstore/
